internal/webserver/webserver.go

```diff
@@ -355,3 +358,27 @@ func (n kubeFilter) getProxyTenantsForOwnerKind(ownerKind capsulev1beta1.OwnerKi
 
 	return
 }
+
+func (n *kubeFilter) sanitizeConnectionHeader(request *http.Request) {
+	headerName := "connection"
+
+	connHeaders := request.Header.Values(headerName)
+
+	var sanitized []string
+
+	for _, values := range connHeaders {
+		parts := strings.Split(values, ",")
+
+		for _, part := range parts {
+			if strings.HasPrefix(strings.ToLower(part), "impersonate-") {
```

Comment:
Instead of removing bad headers, I would suggest following the approach taken by https://github.com/traefik/traefik/pull/8319 to resolve https://github.com/traefik/traefik/security/advisories/GHSA-m697-4v8f-55qg - i.e. handle the connection header directly.

With `if strings.HasPrefix(strings.ToLower(part), "impersonate-") {` you miss edge cases such as (note the leading spaces):

```
curl "https://${API}/api/v1/secrets" -H "Authorization: Bearer ${TOKEN}" -H "Connection: foo, Impersonate-User, Impersonate-Group" -0
```

Reply:
Thanks for the hint!